### DISCUSSION TITLE
Browser contents offset depending on where on the monitor the website is loaded

### DISCUSSION THREAD

**Discussion by derekantrican** [cefsharp/CefSharp#3353] [General]
- What version are you using? (exact version please e.g. 75.1.143, **not** Latest version as that's relative)

CefSharp.WinForms v73.1.130

- Are you using `WPF/WinForms/OffScreen`?

WinForms

- Are you targeting .Net Core? If yes which version?

No

-------------

I have a bit of an odd issue, just wondering if anyone's seen something like this before. It seems that on one monitor on my computer (just one - not both) the browser contents will be offset depending on where the application is located (or - maybe more applicable - where the CefSharp control is located) on the monitor. *The mouse location still seems to trigger the same controls as if the contents were not offset.*

Here's a screen recording: https://youtu.be/oo1UTtblhUs
A couple quick notes on my application:
- The ChromiumWebBrowser control is set to `Dock.Fill` inside the "Player" tab
- For now I've just overridden the "Add Source" button to call `ChromiumWebBrowser.Load("https://www.youtube.com/embed/Y_4i2Oc2zAs");` for testing this issue

It seems to depend on where the browser is "shown". If I click the "Player" tab to show the browser control on the right monitor, then move it to the left monitor, then click the button the content offset can also be seen on the left monitor.

I'm aware I'm on an old version of CefSharp - just wondering if this is some sort of known issue or if things are just super weird on my computer

**Comment by amaitland** [comment]
Do both monitors have the same `DPI`? Is your application `DPI Aware`?

**Comment by amaitland** [comment]
Transitioning to new [Discussions](https://github.com/cefsharp/CefSharp/discussions) section.

**Comment by amaitland** [comment]
Some reason I cannot seem to convert this into a discussion, will check back later and convert when able.

Will be transitioning general questions to `Discussions` here in the main repo in future 😄

**Comment by derekantrican** [comment]
Windows display settings shows the same resolution (3840 x 2160) for each monitor and the same value for "Change the size of text, apps, and other items" (200%). However, the monitors themselves are physically different sizes.

I did a lot of digging on dpi awareness for my application and found that calling `SetProcessDpiAwareness` at the very beginning of my `Program.Main` function with `PROCESS_DPI_AWARENESS.Process_System_DPI_Aware` or `PROCESS_DPI_AWARENESS.Process_Per_Monitor_DPI_Aware` seemed to solve the CEFSharp "location" issue. It causes a lot of other control placement issues, but I think those are on me to figure out :)

```csharp
private enum PROCESS_DPI_AWARENESS
{
    Process_DPI_Unaware = 0,
    Process_System_DPI_Aware = 1,
    Process_Per_Monitor_DPI_Aware = 2
}

[DllImport("SHCore.dll", SetLastError = true)]
private static extern bool SetProcessDpiAwareness(PROCESS_DPI_AWARENESS awareness);
```

Thanks for the help!

**Comment by amaitland** [comment]
You can disable GPU compositing as per https://github.com/cefsharp/CefSharp/wiki/General-Usage#high-dpi-additional-info if making your application DPI aware is problematic. 

> . It causes a lot of other control placement issues, but I think those are on me to figure out :)

.Net 4.7 has a lot of improvements in relation to control sizing/placement, see 
https://github.com/cefsharp/CefSharp/wiki/General-Usage#winforms-high-dpi for link
